app/src/main/java/de/dhbw/corona_world_app/api/Mapper.java

```diff
@@ -105,6 +107,17 @@ private static void initializeMap(API api){
         }
     }
 
+    public static ISOCountry mapISOCodeToISOCountry(String isoCode){
+        if(isoCode.length() != 2) throw new IllegalArgumentException("The given String is no ISOCode! Example:\"DE\"");
+        if(isoCodeToISOCountryMap == null){
+            isoCodeToISOCountryMap = new HashMap<>();
+            for (ISOCountry country:ISOCountry.values()) {
+                isoCodeToISOCountryMap.put(country.getISOCode(), country);
+            }
+        }
```

Comment:
Can be done with parallel stream

Reply:
Perfomance impact of this loop is insignificant, hence using a parallel stream doesn't yield a big advantage. For more info see:
https://stackoverflow.com/questions/20375176/should-i-always-use-a-parallel-stream-when-possible

Reply:
^ (It could even have a negative impact in this case I believe)

Reply:
Did a speedtest, in the average case the For loop approach seems to be 6 times better, but if the creation only happens once, the parallel Stream seems to be the same or a bit better at random.  I would assume the overhead does only play a huge role if the creation is done a lot of times. Either way it seems they are about the same speed with is why i dont mind either way.

Reply:
If you want more performance the fastest way to create This Map is using this approach:
Map<String,ISOCountry> isoCodeToISOCountryMap = new HashMap<>();
            int length=ISOCountry.values().length;
            ISOCountry temp[]=ISOCountry.values();
            for (int i=0;i<length;++i) {
                isoCodeToISOCountryMap.put(temp[i].getISOCode(), temp[i]);
            }
It is about 3/4 of the time your For each has (of course on my pc may be different on an android)